### ISSUE TITLE
Add support for SOCKS proxy

### ISSUE THREAD

**Issue by tuxayo** [ethereum/go-ethereum#1855]
I can't find if Geth support proxies so it's seems that it's not the case.
That would allow to use Geth on restrictive networks for example.

**Comment by karalabe** [comment]
There's no support currently for proxies. It's not really trivial, since the Ethereum discovery protocol runs on UDP and the P2P overlay on TCP, so you need to cross over with both to make geth functional. It would probably be a useful feature, but I'm not sure we have the capacity currently to cater for this use case, but would probably be happy for contributions ;). Though maybe @fjl can pinch in on the topic.

**Comment by fjl** [comment]
SOCKS support is required for users that need/want to tunnel through Tor.
I'll give it a shot for geth 1.5, the release after the next big release.

**Comment by karalabe** [comment]
@fjl I was thinking about a different use case that might or might not impact this. I'm using [Mozilla's Circus](https://circus.readthedocs.org/en/latest/) as a process manager to run my local Ethereum node, have it handle log file rotation, restarts, etc. Among the usual process management features, Circus also has a feature to manage sockets on behalf of the child processes (i.e. when an app needs to listen on a port, circus opens and maintains the socket and the app only "uses" it). This allows both statistics tracking on the network level as well as prevents issues where a fast restart cannot succeed because the TCP port is not yet relinquished to the OS. Just thought it's a cool feature, probably not worth too much effort, but nonetheless :)

**Comment by fjl** [comment]
I'll make a new issue for Socket Activation and related features.

**Comment by djadk84** [comment]
When will this feature will be ready?

**Comment by obscuren** [comment]
Not until after 1.5. 1.4 which is scheduled for "within the next few weeks". 1.5 likely another few.

**Comment by ijcoe6ru** [comment]
Even though we can add socks support to the client, Tor doesn't support UDP. We need a way to use Ethereum without UDP in order to use it over Tor.

**Comment by aphex3k** [comment]
I'm running a full bitcoin node on tor and was wondering if I could do the same for ethereum. From the looks of it, this issue is still open...

**Comment by scarboroughventures** [comment]
I believe this would be very useful for auditing and compliance as well. Being able to intercept outgoing transactions from a server not connected to the client would be very useful to control unauthorized transactions.

**Comment by lelikalf** [comment]
type

**Comment by g3d** [comment]
:+1: here

**Comment by ldlqdsdcn** [comment]
I need this feature

**Comment by ldlqdsdcn** [comment]
Enterprise networks are very complex，For security，Nodes can't access the external network directly

**Comment by Kamilcuk** [comment]
+1 here

**Comment by wxd11a** [comment]
Another +1 here.

**Comment by IamHuskar** [comment]
+1 here

**Comment by he1-sim** [comment]
😊

**Comment by lakshaygaur** [comment]
+1 here

**Comment by ShaunR** [comment]
+1

**Comment by htjain** [comment]
+1

**Comment by aphex3k** [comment]
👀

**Comment by jsvisa** [comment]
👀

**Comment by gibz104** [comment]
👀

**Comment by willianpaixao** [comment]
Related. https://github.com/ethereum/go-ethereum/issues/33345
